source/flux/parts/settings.js

```diff
@@ -88,6 +88,20 @@ export function logInViaCredentials(
     dispatch({type: CREDENTIALS_LOGIN_START})
     const state = getState()
     const isConnected = state.app ? state.app.isConnected : false
+    function showNetworkFailureMessage() {
```

Comment:
Ah. I was thinking just inline the `Alert.alert` calls directly into the spots they're called from:

```js
  if (!isConnected) {
    dispatch({ type: CREDENTIALS_LOGIN_FAILURE });
    Alert.alert(
      "Network Failure",
      "You are not connected to the internet. Please connect if you want to access this feature.",
      [{ text: "OK" }]
    );
  }
  const result = await performLogin(username, password);
  if (result) {
    dispatch({
      type: CREDENTIALS_LOGIN_SUCCESS,
      payload: { username, password }
    });
    // since we logged in successfully, go ahead and fetch the meal info
    dispatch(updateBalances());
  } else {
    dispatch({ type: CREDENTIALS_LOGIN_FAILURE });
    Alert.alert(
      "Invalid Login",
      "The username and password you provided do not match a valid account. Please try again.",
      [{ text: "OK" }]
    );
  }
```

But, I'm OK with having the shorter names to refer to them by.

---

Thus, I'll change my request:

Would you please move these two function definitions back to the top level, and define them above `logInViaCredentials`, please? 

And could you also change them to arrow functions, please? For example, 

```js
const showNetworkFailureMessage = () =>
  Alert.alert(
    "Network Failure",
    "You are not connected to the internet. Please connect if you want to access this feature.",
    [{ text: "OK" }]
  )
```

The extra line taken up by the closing brace just seems wasteful to me, since the function's only one statement long 😜